querygen/querygen.go

```diff
@@ -57,3 +57,26 @@ func GenBuildPipelineRunStartedQuery(index string) string {
 	)
 	return q
 }
+
+// GenBuildPipelineRunCompletedQuery returns a Splunk query for generating Segment events
+// representing success or failure of AppStudio build PipelineRuns.
+func GenBuildPipelineRunCompletedQuery(index string) string {
+	q, _ := UserJourneyQueryGen(
+		index,
+		`verb=update `+
+			`"responseStatus.code"=200 `+
+			`"objectRef.apiGroup"="tekton.dev" `+
+			`"objectRef.resource"="pipelineruns" `+
+			`"objectRef.subresource"="status" `+
+			`"responseObject.metadata.labels.pipelines.appstudio.openshift.io/type"=build `+
+			`"responseObject.metadata.resourceVersion"="*" `+
+			`"responseObject.status.startTime"="*" `+
+			`| spath path=responseObject.status.conditions{} output=conditions `+
+			`| mvexpand conditions `+
+			`| spath input=conditions `+
```

Comment:
Since this is very similar to the existing query in GenBuildPipelineRunStartedQuery, could both be combined?

The trouble with `spath` is that you can't see the fields that are being generated in the query string - so things are less readable. Perhaps we could give up on the first `spath` and run `mvexpand` directly on the nested field?

As for the 2nd `spath` could we do without it? As far as I can tell, it would make the "where" expression a bit longer and we would need an eval to generate "reason" and "message" which would be clearer IMO.

Also consider using "search" rather then where to keep things consistent - as long as you don't need any eval functions, search should be sufficient.

Reply:
I made some suggestions on https://github.com/redhat-appstudio/segment-bridge/pull/36#discussion_r1304940201 related to these comments. PTAL and maybe we can incorporate them here.